app/forms/course_years_form.rb

```diff
@@ -9,25 +9,45 @@ class CourseYearsForm
 
   validates :course_year, inclusion: { in: ->(rec) { rec.course_years_options.keys } }
 
-  def initialize(params = {})
+  def initialize(trainee: nil, params: {})
     assign_attributes(params)
+    @trainee = trainee
   end
 
   def course_year
     @course_year.to_i
   end
 
+  # { 2022=>"2022 to 2023", 2021 => "2021 to 2022" }
   def course_years_options
+    course_year_to.downto(course_year_from).map.index_with { |year| "#{year} to #{year + 1}" }
+  end
+
+private
+
+  def course_year_from
     [
-      default_course_year,
-      default_course_year - 1,
-    ].inject({}) do |sum, y|
-      sum[y] = "#{y} to #{y + 1}"
-      sum
-    end
+      recruitment_cycle_year,
+      itt_start_date_year,
+      course_year_to - 1,
+    ].compact.min
```

Comment:
Can you explain what this bit is doing? the previous felt easier to read - by default (drafts) we want the form to show the default course year and the one prior - I don't think that's the same as the recruitment cycle year.

Reply:
I think in our new form, we'd want the available years to be all the years between the default course year and the "start year" - where the start year is the earlier of itt start date or the existing course year.